pkg/controller/podautoscaler/horizontal_test.go

```diff
@@ -1720,4 +1720,89 @@ func TestAvoidUncessaryUpdates(t *testing.T) {
 	tc.runTestWithController(t, controller, informerFactory)
 }
 
+func TestConvertDesiredReplicasWithRules(t *testing.T) {
+	conversionTestCases := []struct {
+		currentReplicas                  int32
+		desiredReplicas                  int32
+		hpaMinReplicas                   int32
+		hpaMaxReplicas                   int32
+		expectedConvertedDesiredReplicas int32
+		annotation                       string
+	}{
+		{
+			currentReplicas:                  5,
+			desiredReplicas:                  7,
+			hpaMinReplicas:                   3,
+			hpaMaxReplicas:                   8,
+			expectedConvertedDesiredReplicas: 7,
+			annotation:                       "prenormalized desired replicas within range",
+		},
+		{
+			currentReplicas:                  3,
+			desiredReplicas:                  1,
+			hpaMinReplicas:                   2,
+			hpaMaxReplicas:                   8,
+			expectedConvertedDesiredReplicas: 2,
+			annotation:                       "prenormalized desired replicas < minReplicas",
+		},
+		{
+			currentReplicas:                  1,
+			desiredReplicas:                  0,
+			hpaMinReplicas:                   0,
+			hpaMaxReplicas:                   10,
+			expectedConvertedDesiredReplicas: 1,
+			annotation:                       "1 is minLimit because hpaMinReplicas < 1",
+		},
+		{
+			currentReplicas:                  20,
+			desiredReplicas:                  1000,
+			hpaMinReplicas:                   1,
+			hpaMaxReplicas:                   10,
+			expectedConvertedDesiredReplicas: 10,
+			annotation:                       "maxReplicas is the limit because maxReplicas < scaleUpLimit",
+		},
+		{
+			currentReplicas:                  3,
+			desiredReplicas:                  1000,
+			hpaMinReplicas:                   1,
+			hpaMaxReplicas:                   2000,
+			expectedConvertedDesiredReplicas: calculateScaleUpLimit(3),
+			annotation:                       "scaleUpLimit is the limit because scaleUpLimit < maxReplicas",
+		},
+	}
+
+	for _, ctc := range conversionTestCases {
+		actualConvertedDesiredReplicas, err := convertDesiredReplicasWithRules(
+			ctc.currentReplicas, ctc.desiredReplicas, ctc.hpaMinReplicas, ctc.hpaMaxReplicas,
+		)
+
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
```

Comment:
don't Fatalf here, just continue

Reply:
Ah yes good point - the test will just fail anyway.